server/controller/apis/topicController.js

```diff
@@ -278,33 +247,83 @@ exports.saveTopic = async ( req, res, redirect ) => {
 
         if(topic.topicType == 3) {
             
-            topic.topicContentHtml = req.body.embedded_submission_text_topic;
+            topic.topicHtml = req.body.embedded_submission_text_topic;
         }
         else {
             // check to see if the incomping message format is from the UI or the API
-            if( req.body.topicContentHtml ) {
-                topic.topicContentHtml = req.body.topicContentHtml;
+            if( req.body.topicHtml ) {
+                topic.topicHtml = req.body.topicHtml;
             }
             else {
-                topic.topicContentHtml = req.body.topicEditor;
+                topic.topicHtml = req.body.topicEditor;
             }
         }
-        topic.topicLink = req.body.topicLink;
         
         // check to see if the incoming message format is from the UI form or the API
-        if( req.body.topicActive ) {
-            topic.active = ( req.body.topicActive == "on" ) ? true : false;
-        }
-        else if ( req.body.active ) {
+        topic.active = false; // Defaulted to false if not specified.
+        if( req.body.active ) {
             topic.active = req.body.active;
         }
-        
-        topic.isRequired = ( req.body.isRequired == "on" || req.body.isRequired == true ) ? true : false;
```

Comment:
Based on @briangormanly 's feedback in class today, we should leave this intact and adjust at a later time if needed.